### PR TITLE
Outgoing response created in worker

### PR DESCRIPTION
**Pull request by mklander**
A suggestion to simplify the outgoing publisher - pass the required message in from the worker rather than passing a function to build it.

### CODE REVIEW

File: common/src/test/scala/uk/ac/wellcome/platform/archive/common/generators/BagRequestGenerators.scala [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
     BagRequest(
       requestId = randomUUID,
       bagLocation = bagLocation
     )
+
+  def createBagLocation() =
+    BagLocation(
+      storageNamespace = randomAlphanumeric(),
+      storagePrefix = Some(randomAlphanumeric()),
+      storageSpace = StorageSpace(randomAlphanumeric()),
```
Comment (alexwlchan) [inline common/src/test/scala/uk/ac/wellcome/platform/archive/common/generators/BagRequestGenerators.scala]: nit: I think there's a helper like `createStorageSpace`.

Comment (mklander) [reply common/src/test/scala/uk/ac/wellcome/platform/archive/common/generators/BagRequestGenerators.scala]: Thanks, so many Generators

File: common/src/main/scala/uk/ac/wellcome/platform/archive/common/operation/OutgoingPublisher.scala [hunk 2]
```diff
@@ -1,42 +1,29 @@
...
-  )(implicit
+  def sendIfSuccessful[R, O](result: OperationResult[R], outgoing: => O)(
+    implicit
     ec: ExecutionContext,
-    enc: Encoder[O]): Future[Unit] = result match {
-    case OperationSuccess(summary) =>
-      sendOutgoing(transform(summary)).map(_ => ())
-    case OperationFailure(_, _) =>
+    enc: Encoder[O]): Future[Unit] = {
+    if (result.isSuccessful) {
```
Comment (kenoir) [inline common/src/main/scala/uk/ac/wellcome/platform/archive/common/operation/OutgoingPublisher.scala]: You can replace this with a match and `OR` the two success type together - that way you can avoid the case to boolean mapping below.

File: common/src/main/scala/uk/ac/wellcome/platform/archive/common/ingests/operation/OperationResult.scala [hunk 3]
```diff
@@ -3,6 +3,16 @@ package uk.ac.wellcome.platform.archive.common.ingests.operation
 sealed trait OperationResult[T] {
   val summary: T
   def withSummary(summary: T): OperationResult[T]
+  def isSuccessful: Boolean = {
```
Comment (kenoir) [inline common/src/main/scala/uk/ac/wellcome/platform/archive/common/ingests/operation/OperationResult.scala]: This is _not_ very scala-esque.

**Pull request merged**